Entity/Currency.php

```diff
@@ -9,4 +9,13 @@
  */
 abstract class Currency extends BaseCurrency
 {
+    public function setCode($code)
+    {
+        $this->code = $code;
```

Comment:
must contains return $this;

To simplify such decorations: 

$currency->setCode()->setSomethingElse()